### PR TITLE
fix: "linux-arm64-openssl-undefined" is now "linux-arm64-openssl-1.1.x"

### PR DESCRIPTION
**Pull request by jkomyno**
Contributes https://github.com/prisma/prisma/issues/16232

`libssl` can be undefined when the following blocks are run:

```typescript
if (platform === 'linux' && arch === 'arm64') {
  // 64 bit ARM
  return `linux-arm64-openssl-${libssl}` as Platform
}

if (platform === 'linux' && arch === 'arm') {
  // 32 bit ARM
  return `linux-arm-openssl-${libssl}` as Platform
}
```

which could happen e.g. in `node:lts-buster-slim` / `node:16-slim`, as Debian Buster doesn't come with any `libssl` installed by default. This fix at least tells the uses which `binaryTarget` Prisma tried to download, but Prisma will likely still fail at runtime if `libssl` is actually missing from the system.

### CODE REVIEW

File: packages/get-platform/src/getPlatform.ts [hunk 1]
```diff
@@ -282,7 +287,7 @@ export async function getPlatform(): Promise<Platform> {
 
   // use the debian build with OpenSSL 1.1 as a last resort
   // TODO: perhaps we should default to 'debian-openssl-3.0.x'
-  return 'debian-openssl-1.1.x'
+  return `debian-openssl-${defaultLibssl}`
```
Comment (Jolg42) [inline packages/get-platform/src/getPlatform.ts:290]: Maybe this makes the change a bit clearer for the untouched code?
```diff
  // if just OpenSSL is known, fallback to debian with a specific libssl version
  if (osResult.libssl) {
-    return ('debian-openssl-' + osResult.libssl) as Platform
+    return ('debian-openssl-' + osResult.libssl) as Platform
  }
```

**Review by Jolg42 — APPROVED** [review]
Looks good 👍🏼 

(See comment before merging)

File: packages/get-platform/src/getPlatform.ts (hunk 1 above)
Comment (jkomyno) [reply packages/get-platform/src/getPlatform.ts:308]: I've followed a similar advice to make the default more explicit, see https://github.com/prisma/prisma/pull/17151/commits/6b9dfb64b7bc632a174ab2b0ada71f70c2132b1f

**Review by janpio — CHANGES_REQUESTED** [review]
Internal discussion: https://prisma-company.slack.com/archives/C04DG9JL6TV/p1672933294609189?thread_ts=1672926697.271059&cid=C04DG9JL6TV

**Comment by jkomyno** [comment]
Note: the code of this PR was included in https://github.com/prisma/prisma/pull/17334, which got merged for `prisma@4.9.0`.

**Pull request closed without merging**